### PR TITLE
fixed lbaas flag for azure

### PR DESCRIPTION
**Pull request by fsniper**

### CODE REVIEW

**Comment by afritzler** [comment]
@fsniper sorry for the huge delay here. Can you resolve the merge conflicts please. (also for #42) Thx!

File: platforms/azure/modules/machine/machine.tf [hunk 1]
```diff
@@ -37,6 +37,8 @@ locals {
   offer = "${element(split("${local.separator}",local.image),1)}"
   sku = "${element(split("${local.separator}",local.image),2)}"
   version = "${element(split("${local.separator}",local.image),3)}"
+
+  public_key = ""
```
Comment (mandelsoft) [inline platforms/azure/modules/machine/machine.tf]: Again this strange unused variable

File: platforms/azure/modules/lbaas/lbaas.tf [hunk 2]
```diff
@@ -91,17 +98,17 @@ resource "azurerm_lb_probe" "lb" {
...
-  value = "${azurerm_public_ip.lb.ip_address}"
+  value = "${data.azurerm_public_ip.lb.*.ip_address}"
 }
 output "vip_type" {
   value = "A"
 }
 
 locals {
-  pool_ids = "${compact(concat(azurerm_lb_backend_address_pool.lb.*.id,list("")))}"
+  pool_ids = "${compact(concat(azurerm_lb_backend_address_pool.lb.*.id,list("")))}"
```
Comment (mandelsoft) [inline platforms/azure/modules/lbaas/lbaas.tf]: ????

File: platforms/azure/modules/lbaas/lbaas.tf [hunk 3]
```diff
@@ -47,6 +47,13 @@ resource "azurerm_public_ip" "lb" {
   public_ip_address_allocation = "Static"
 }
 
+data "azurerm_public_ip" "lb" {
```
Comment (mandelsoft) [inline platforms/azure/modules/lbaas/lbaas.tf]: Could explain, what this is for. You know Azure for sure better then me. Is this special for the azurerm. Why do you need the data if the ip is generated by a regular resource?

File: platforms/azure/modules/lbaas/lbaas.tf (hunk 2 above)
Comment (fsniper) [reply platforms/azure/modules/lbaas/lbaas.tf]: Oh, extra whitespace. This should be removed of course.

File: platforms/azure/modules/machine/machine.tf (hunk 1 above)
Comment (fsniper) [reply platforms/azure/modules/machine/machine.tf]: I don't remember the reason for this, but If I recall correctly, this ensured the apply. I think it was requesting a public_key.

File: platforms/azure/modules/lbaas/lbaas.tf (hunk 3 above)
Comment (fsniper) [reply platforms/azure/modules/lbaas/lbaas.tf]: in azure, public ip's are allocated and set after the resource is created. So Terraform won't have the knowledge of this public ip at deploy time. This makes it impossible to deploy with a public ip. To solve this, there is this public_ip resource. This ensures the ip is allocated before the wanted resource is deployed.

File: platforms/azure/modules/machine/machine.tf (hunk 1 above)
Comment (mandelsoft) [reply platforms/azure/modules/machine/machine.tf]: Long ago....  I just tried to figure out what is going on here. It is a little bit more complicate:

The public key is required to create machines on Azure. Therefore the MachineClass for Azure (used by the Machine Controller to dynamically manage worke machines) also requires this info.

There is a local varible initialized with "" so far. It should take

`"${lookup(var.vm_info,"public_key","")}"`

Then the template var should use this varaible `${local.public_key}`

To make this work the vm module must export this key in azure/modules/vms/vms.tf (output vm_info). Add the key
`public_key = "${var.key}"`

Do you want to do this to make youre correction complete, or should I prepare a commit for this.

Unfortunatley we don't use the azure part anymore. Therefore there are often just formal change. Sorry for that.

File: platforms/azure/modules/lbaas/lbaas.tf (hunk 3 above)
Comment (mandelsoft) [reply platforms/azure/modules/lbaas/lbaas.tf]: Yepp I joust thought somethinglike this. The Azurerm provider is not really convincing. There are so many problems.

We could not solve the major problem: handling of attached disks for the vm resource. If used incrementally (not only for creation) the resource does not handle dynamically attached resources (by the kubernetes controllers to persistent volumes). As a result it is NOT possible to
do day-2 operations (kubify was designed for) on Azure.

We tried to address this problem at kubecon, but we couldn't find a valid contact.

**Comment by mandelsoft** [comment]
I've done the fixes mentioned in the review comment above on the master branch. Then I tried to merge using the 'Update branch' button above. Github states, that everything is ok, but the merge commit
does nothing than ignore my changes. Basically the state now is lying.
You should incorporate you changes on to of the master branch inclusing the changes done there.
I cannot do this.

**Comment by mandelsoft** [comment]
I've cleaned the history a little bit

**Pull request merged**